### PR TITLE
[Fix](Nereids) fix leading with multi level of brace pairs

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

fix leading with multi level of brace pairs
example:
leading(t1 {{t2 t3} {t4 t5}} t6) can be reduced to leading(t1 {t2 t3 {t4 t5}} t6)
also update cases which remove project node from explain shape plan 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java [hunk 1]
```diff
@@ -124,6 +124,32 @@ public LeadingHint(String hintName, List<String> parameters, String originalStri
             }
             lastParameter = parameter;
         }
+        normalizeLevelList();
+    }
+
+    private void removeGap(int left, int right, int gap) {
+        for (int i = left; i <= right; i++) {
+            levellist.set(i, levellist.get(i) - (gap - 1));
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:132]: levellist => levelList

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java [hunk 2]
```diff
@@ -124,6 +124,32 @@ public LeadingHint(String hintName, List<String> parameters, String originalStri
...
+        }
+    }
+
+    // when leading extract levellist like 0 2 2 3 3 0, it could be reduced to 0 1 1 2 2 0
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:136]: please define what is the “gap”
it is better to copy the example in "Proposed Change" to illustrate the concept of gap

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:132]: get

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java (hunk 2 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:136]: get

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:136]: gap is like 0 to 2 in upper example, in this view we can see the represent of gap in levelList level

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**